contracts/mechanics/garden-of-forking-paths/GardenOfForkingPaths.sol

```diff
@@ -38,9 +43,12 @@ The reward must be a Terminus token and the Garden of Forking Paths contract mus
 on the token pool.
  */
 struct Reward {
-    address terminusAddress;
-    uint256 terminusPoolId;
+    uint256 rewardType; // 1, 1155, 20, 721 - 1 means mint Terminus, 1155 means transfer 1155
+    address rewardAddress;
+    uint256 rewardTokenID;
     uint256 rewardAmount;
+    address inventoryAddress; // if 0, reward goes to player/staker, else to NFT
+    uint256 inventorySlot;
```

Comment:
What will the behavior be if the inventory slot already has an item equipped in it?

Two cases:
1. Slot is persistent - allow NFT to proceed without the reward?
2. Slot is impersistent - should we unequip and send the item back to the staker?